openshift-kube-apiserver/admission/customresourcevalidation/authentication/validate_authentication.go

```diff
@@ -121,14 +147,122 @@ func validateAuthenticationSpec(spec configv1.AuthenticationSpec) field.ErrorLis
 				spec.WebhookTokenAuthenticator, fmt.Sprintf("this field cannot be set with the %q .spec.type", spec.Type),
 			))
 		}
-
 	}
 
 	errs = append(errs, crvalidation.ValidateConfigMapReference(specField.Child("oauthMetadata"), spec.OAuthMetadata, false)...)
 
-	return errs
+	// Perform External OIDC Provider related validations
+	// ----------------
+
+	// There is currently no guarantee that these fields are not set when the spec.Type is != OIDC.
+	// To ensure we are enforcing approriate admission validations at all times, just always iterate through the list
+	// of OIDC Providers and perform the validations.
+	// If/when the openshift/api admission validations are updated to enforce that this field is not configured
+	// when Type != OIDC, this loop should be a no-op due to an empty list.
+	for i, provider := range spec.OIDCProviders {
+		err, warnings := validateOIDCProvider(specField.Child("oidcProviders").Index(i), cel, provider)
+		errs = append(errs, err...)
+		warns = append(warns, warnings...)
+	}
+	// ----------------
+
+	return errs, warns
 }
 
 func validateAuthenticationStatus(status configv1.AuthenticationStatus) field.ErrorList {
 	return crvalidation.ValidateConfigMapReference(field.NewPath("status", "integratedOAuthMetadata"), status.IntegratedOAuthMetadata, false)
 }
+
+func validateOIDCProvider(path *field.Path, cel celStore, provider configv1.OIDCProvider) (field.ErrorList, []string) {
+	return validateClaimMappings(path, cel, provider.ClaimMappings)
+}
+
+func validateClaimMappings(path *field.Path, cel celStore, claimMappings configv1.TokenClaimMappings) (field.ErrorList, []string) {
+	path = path.Child("claimMappings")
+	errs := field.ErrorList{}
+	warns := []string{}
+
+	err, warn := validateUIDClaimMapping(path, cel, claimMappings.UID)
+	errs = append(errs, err...)
+	warns = append(warns, warn)
```

Comment:
Should this be happening if warn is empty?

Reply:
It should not - will update.